src/substantial/src/converters.rs

```diff
@@ -125,6 +239,97 @@ impl Run {
     pub fn reset(&mut self) {
         self.operations = vec![];
     }
+
+    pub fn next_id(&mut self) -> u32 {
+        self.id += 1;
+        self.id
+    }
```

Comment:
I suggest creating a context struct that owns a run and move that into another file, just like the way `deno_context.ts` is designed.